engine/profiler/src/profiler_proc_utils.cpp

```diff
@@ -0,0 +1,129 @@
+// Copyright 2020-2022 The Defold Foundation
```

Comment:
Now I definitely think that these two tasks should be split up in two PR's (android memory usage vs JNI env profile property)

Reply:
My I wasn't clear in my explanations.

It's just a refactoring. I moved work with `proc` from `profiler_linux` into `profiler_proc_utils` to be able to create `profiler_android` and use proc from both android and linux

Reply:
Sure, but the refactoring isn't related to the android thread profiler variable.

Reply:
Most of the code in this PR is about the refactoring, yet the title says something different.

Reply:
yes, it's just because it's hard to implement andoid related functionality without this refactoring